oauth2/src/main/java/com/linecorp/armeria/client/auth/oauth2/OAuth2ClientCredentialsGrantBuilder.java

```diff
@@ -0,0 +1,156 @@
+
+/**
+ * Builds {@link OAuth2ClientCredentialsGrant}.
+ */
+public class OAuth2ClientCredentialsGrantBuilder extends AbstractOAuth2AuthorizationGrantBuilder {
+
+    /**
+     * A common abstraction for the requests implementing various Access Token request/response flows,
+     * as per <a href="https://tools.ietf.org/html/rfc6749">[RFC6749]</a>.
+     * @param accessTokenEndpoint A {@link WebClient} to facilitate an Access Token request. Must correspond to
+     *                            the Access Token endpoint of the OAuth 2 system.
+     * @param accessTokenEndpointPath A URI path that corresponds to the Access Token endpoint of the
+     */
+    OAuth2ClientCredentialsGrantBuilder(WebClient accessTokenEndpoint, String accessTokenEndpointPath) {
+        super(accessTokenEndpoint, accessTokenEndpointPath);
+    }
+
+    /**
+     * Provides client authorization for the OAuth 2.0 requests based on encoded authorization token and
+     * authorization type,
+     * as per <a href="https://tools.ietf.org/html/rfc6749#section-2.3">[RFC6749], Section 2.3</a>.
+     *
+     * @param authorizationSupplier A supplier of encoded client authorization token.
+     * @param authorizationType One of the registered HTTP authentication schemes as per
+     *                          <a href="https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml">
+     *                          HTTP Authentication Scheme Registry</a>.
+     */
+    @Override
+    protected OAuth2ClientCredentialsGrantBuilder clientAuthorization(
+            Supplier<String> authorizationSupplier, String authorizationType) {
+        super.clientAuthorization(authorizationSupplier, authorizationType);
+        return this;
+    }
+
+    /**
+     * Provides client authorization for the OAuth 2.0 requests based on encoded authorization token and
+     * {@code Basic} authorization type,
+     * as per <a href="https://tools.ietf.org/html/rfc6749#section-2.3">[RFC6749], Section 2.3</a>.
+     *
+     * @param authorizationSupplier A supplier of encoded client authorization token.
+     */
+    @Override
+    protected OAuth2ClientCredentialsGrantBuilder clientBasicAuthorization(
+            Supplier<String> authorizationSupplier) {
+        super.clientBasicAuthorization(authorizationSupplier);
+        return this;
+    }
+
+    /**
+     * Provides client authorization for the OAuth 2.0 requests based on client credentials and
+     * authorization type,
+     * as per <a href="https://tools.ietf.org/html/rfc6749#section-2.3">[RFC6749], Section 2.3</a>.
+     *
+     * @param credentialsSupplier A supplier of client credentials.
+     * @param authorizationType One of the registered HTTP authentication schemes as per
+     *                          <a href="https://www.iana.org/assignments/http-authschemes/http-authschemes.xhtml">
+     *                          HTTP Authentication Scheme Registry</a>.
+     */
```

Comment:
Global comment: You don't need to copy the Javadoc from the superclass if you did not change anything. Javadoc tool will do that for you.

Reply:
removed those methods altogether